src/components/tl/cuda/allgatherv/allgatherv_linear.c

```diff
@@ -218,19 +220,21 @@ ucc_tl_cuda_allgatherv_linear_progress_frag(ucc_tl_cuda_task_t *task)
             } else {
                 sbuf = PTR_OFFSET(task->allgatherv_linear.sbuf, frag_offset);
             }
-            for (i = 0; i < tsize; i++) {
-                if (i == trank) {
-                    continue;
-                }
+            eargs.task_type =  UCC_EE_EXECUTOR_TASK_COPY_MULTI;
```

Comment:
the fields `num_vectors` and `task_type` of `eargs.copy_multi` could be filled once before the branching "if (step == 0) {" line 209